### PR TITLE
1 - WebScript to add single or more than one member to a group

### PR DESCRIPTION
**Pull request by DarkStar1**

### CODE REVIEW

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import org.json.JSONException;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.springframework.extensions.webscripts.*;
+
+import java.io.IOException;
+import java.util.*;
+
+public class Members extends AbstractWebScript {
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java:21]: "Members" is a domain object name. Name of a class should define what it is. "MembersWebScript" would be much better

File: openesdh-repo/src/main/amp/config/alfresco/module/openesdh-repo/context/openesdh-repository-context.xml [hunk 2]
```diff
@@ -288,77 +303,81 @@
...
-        <property name="personService" ref="PersonService"/>
-        <property name="dictionaryService" ref="DictionaryService"/>
-        <property name="authenticationService" ref="AuthenticationService"/>
-        <property name="workflowService" ref="WorkflowService"/>
+    <bean id="webscript.dk.openesdh.documents.categories.doccategory.get" parent="documentCategoriesWebscript"/>
+    <bean id="webscript.dk.openesdh.documents.categories.doccategory.post" parent="documentCategoriesWebscript"/>
+    <bean id="webscript.dk.openesdh.documents.categories.doccategory.delete" parent="documentCategoriesWebscript"/>
+
+    <!--Adding members to group-->
+    <bean id="webscript.org.alfresco.repository.groups.children.put" parent="webscript"
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/amp/config/alfresco/module/openesdh-repo/context/openesdh-repository-context.xml:365]: we now have a superb annotations to implement webscripts. you should use it

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    private AuthorityService authorityService;
+    private PersonService personService;
+    // Logger
+    private static final Log logger = LogFactory.getLog(Members.class);
+
+    @Override
+    public void execute(WebScriptRequest req, WebScriptResponse res) throws IOException {
+        String method = req.getServiceMatch().getWebScript().getDescription().getMethod();
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java:30]: ~~variable is not used anywhere~~
sorry found it :)

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+    private void addMembers(WebScriptRequest req, WebScriptResponse res, String groupShortName, JSONObject parsedRequest) throws IOException, JSONException {
+        String users = getOrNull(parsedRequest, "users");
+        String groups = getOrNull(parsedRequest, "groups");
+        StringBuffer failed = new StringBuffer();
+        String groupFullName ="GROUP_"+ groupShortName;
+
+        if (StringUtils.isNotBlank(users)) {
+            try {
+                List<String> candidates = Arrays.asList(StringUtils.split(users, ","));
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java:68]: have you considered parsing this param to JSONArray?
and on current situation it is not necessary to convert to list, you can iterate over array

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            } catch (Exception ge) {
+                logger.warn("**** could not add users to group:\n\t\t\t" + ge.getMessage() + " ****");
+            }
+        }
+
+        if (StringUtils.isNotBlank(groups)) {
+            try {
+                List<String> candidates = Arrays.asList(StringUtils.split(groups, ","));
+                for (String group : candidates) {
+                    String memberFullName = "GROUP_"+ group;
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/groups/Members.java:86]: "GROUP_" should be extracted as constant

**Comment by DarkStar1** [comment]
retest this please

**Comment by DarkStar1** [comment]
please retest this

**Pull request merged**